commons/src/main/java/io/aiven/kafka/tiered/storage/commons/objectstorage/ObjectStorageFactory.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2023 Aiven Oy
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.aiven.kafka.tiered.storage.commons.objectstorage;
+
+import org.apache.kafka.common.Configurable;
+
+public interface ObjectStorageFactory extends Configurable {
```

Comment:
The main reason is that it would be nice to separate upload and download paths, let them have different instances of (S3 and other) clients with potentially different settings for performance reasons.

Reply:
Oh that's fair. Maybe we can then have 2 instead of 3 abstractions then? 
I think I've heard "archiver" and "fetcher" in some related discussions. Maybe those are good ones to group these?

- SecondTierFetcher <- FileFetcher
- SecondTierArchiver <- FileUploader + FileDeleter